src/appshell/view/dockwindow_kdab/dockwindow.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "dockwindow.h"
+
+#include "dockpage.h"
+#include "docktoolbar.h"
+
+#include "log.h"
+
+#include "thirdparty/KDDockWidgets/src/MainWindowBase.h"
+#include "thirdparty/KDDockWidgets/src/DockWidgetQuick.h"
+
+namespace mu::dock {
+class MainWindow : public KDDockWidgets::MainWindowBase
+{
+public:
+    MainWindow(QQuickItem* parent)
+        : KDDockWidgets::MainWindowBase("mainWindow", KDDockWidgets::MainWindowOption_None, parent)
+    {
+    }
+
+private:
+    KDDockWidgets::SideBar* sideBar(KDDockWidgets::SideBarLocation) const override
+    {
+        return nullptr;
+    }
+
+    QMargins centerWidgetMargins() const override
+    {
+        return {};
+    }
+};
+}
+
+using namespace mu::dock;
+
+DockWindow::DockWindow(QQuickItem* parent)
+    : QQuickItem(parent),
+    m_mainWindow(new MainWindow(this)),
+    m_toolBars(this),
+    m_pages(this)
+{
+}
+
+QString DockWindow::currentPageUri() const
+{
+    return m_currentPageUri;
+}
+
+QQmlListProperty<mu::dock::DockToolBar> DockWindow::toolBarsProperty()
+{
+    return m_toolBars.property();
+}
+
+QQmlListProperty<mu::dock::DockPage> DockWindow::pagesProperty()
+{
+    return m_pages.property();
+}
+
+void DockWindow::loadPage(const QString& uri)
+{
+    TRACEFUNC;
+
+    if (m_currentPageUri == uri) {
+        return;
+    }
+
+    DockPage* newPage = pageByUri(uri);
+    IF_ASSERT_FAILED(newPage) {
+        return;
+    }
+
+    DockPage* currentPage = pageByUri(m_currentPageUri);
+    if (currentPage) {
+        currentPage->close();
+    }
+
+    newPage->init(*this);
+
+    DockToolBar* prevToolBar = nullptr;
+    for (DockToolBar* toolBar : m_toolBars.list()) {
+        auto location = prevToolBar ? KDDockWidgets::Location_OnRight : KDDockWidgets::Location_OnTop;
+        addDock(toolBar, location, prevToolBar);
+
+        if (!toolBar->allowedPagesUriList().empty() && !toolBar->allowedPagesUriList().contains(uri)) {
+            toolBar->close();
+        }
```

Comment:
Confusing.
As far as I can see from the code (and as it was before), we have:
1. Toolbars of the window - always shown
2. Page toolbars - shown if the page is shown

It is strange to see the window toolbar that needs to be hidden for the given pages. Maybe then such a toolbar can simply be move to the page?

Reply:
fixed